airgun/views/computeresource.py

```diff
@@ -199,34 +215,207 @@ class compute_profiles(SatTab):
+    """Compute Resource profile "Network interface" item widget"""
+    name = TextInput(locator=".//input[contains(@id, 'name')]")
+    network = FilteredDropdown(id="network")
+    interface_type = FilteredDropdown(id="interface")
+
+
+class ComputeResourceRHVProfileStorageItem(GenericRemovableWidgetItem):
+    """Compute Resource profile "Storage" item widget"""
+    size = TextInput(locator=".//input[contains(@id, 'size_gb')]")
+    storage_domain = FilteredDropdown(id="storage_domain")
+    preallocate_disk = Checkbox(locator=".//input[contains(@id, 'preallocate')]")
+    wipe_disk_after_delete = Checkbox(locator=".//input[contains(@id, 'wipe_after_delete')]")
+    disk_interface = FilteredDropdown(id="interface")
+
+    @View.nested
+    class bootable(View):
+        ROOT = ".//input[contains(@id, 'bootable')]"
+
+        def _is_checked(self):
+            return self.browser.get_attribute('checked', self)
+
+        def read(self):
+            return self.browser.is_selected(self)
+
+        def fill(self, value):
+            if value is True and not self.browser.is_selected(self):
+                self.browser.click(self)
+
+
+def _is_provider(provider_type, current_provider):
+    """Check if the current provider is of type provider_type, used with ConditionalSwitchableView.
+    The view must contain an attribute current_provider that will be read and returned by
+    ConditionalSwitchableView instance, see ResourceProviderProfileView class for usage.
+
+    Note: The provider type is always appended to the end of the compute resource name,
+    for example: compute resource name "foo"
+
+        1. For RHV type the provider name will be displayed as: "foo (RHV)"
+        2. For EC2 type the provider name will be dipalayed as: "foo (ca-central-1-EC2)" where
+            "ca-central-1" is the region.
+    """
+    return current_provider.endswith('{0})'.format(provider_type))
+
 
 class ResourceProviderProfileView(BaseLoggedInView):
     breadcrumb = BreadCrumb()
-    compute_profile = FilteredDropdown(
-        id='s2id_compute_attribute_compute_profile_id')
-    compute_resource = FilteredDropdown(
-        id='s2id_compute_attribute_compute_resource_id')
-    flavor = FilteredDropdown(id='s2id_compute_attribute_vm_attrs_flavor_id')
-    image = FilteredDropdown(id='s2id_compute_attribute_vm_attrs_image_id')
-    availability_zone = FilteredDropdown(
-        id='s2id_compute_attribute_vm_attrs_availability_zone')
-    subnet = FilteredDropdown(id='s2id_compute_attribute_vm_attrs_subnet_id')
-    security_groups = MultiSelect(
-        id='ms-compute_attribute_vm_attrs_security_group_ids')
-    managed_ip = FilteredDropdown(
-        id='s2id_compute_attribute_vm_attrs_managed_ip')
+    compute_profile = FilteredDropdown(id='s2id_compute_attribute_compute_profile_id')
+    compute_resource = FilteredDropdown(id='s2id_compute_attribute_compute_resource_id')
+
+    provider_content = ConditionalSwitchableView()
```

Comment:
Why such complex logic with `_is_provider` and partial calls instead of ConditionalSwitchableView(reference='current_provider') and extending current_provider to properly grab only provider part?

Reply:
we have not to change that provider value when reading it, as it may be used by some checking reading tests , remain variant to use a callable.

Reply:
using a callable is allowed by the API it self.

Reply:
also do not want to use regex to extract that provider that can lead to errors, the way it's implemented now will work value that end with "provider)".

Reply:
No need to change value, yea, i was suggesting using regex. Why do you think it will lead to errors?
As a last resort option you can use `rsplit('(', 1)` to have the same effect. 
Imo current implementation is too tricky and hard to understand from the first glance.

Reply:
@abalakh will see what I do do here.

Reply:
Done via regex